### ISSUE TITLE
fix openSSL cross-platform build issues

### ISSUE THREAD

**Issue by yoshuawuyts** [yoshuawuyts/crossgen#7]
# Bug Report
## Your Environment
| Software         | Version(s) |
| ---------------- | ---------- |
| crossgen      |0.2.0
| Rustc            |1.30.0 - nightly
| Operating System | Linux

## Expected Behavior
All builds on Travis should pass.

## Current Behavior
OpenSSL doesn't build on armv7 and BSD due to
https://github.com/sfackler/rust-openssl/issues/977. We either need to include a vendored version of openssl, or switch to RusTLS.

**Comment by yoshuawuyts** [comment]
Asked a question about it on stackoverflow: https://stackoverflow.com/questions/52116642/what-is-the-right-way-to-encrypt-data-using-pkcs1-using-rustls

**Comment by soyuka** [comment]
IMHO would be better to add vendored versions of openssl to these platforms (or at least make it a requirement). It may be given as additional documented steps required to make openssl work?

**Comment by yoshuawuyts** [comment]
@soyuka oh interesting - why do you think a vendored version of OpenSSL would work better than RusTLS?

I'm personally leaning slightly more towards the latter. Mostly because it seems like it would require the least amount of work to compile cross-platform. But it'd be great to hear your thoughts on this!

**Comment by soyuka** [comment]
I totally agree that, from a cross-platform point of view, rustls would be a really good thing. But, OpenSSL has many years of existence and stability whereas rustls is:

> currently in development and hence unstable

I've been thinking though, if this is only a matter of encrypting secrets for deployments, it might not be that of an issue. Still, regarding maintenance, as rustls isn't really stable it may be more work in the future to keep this library up to date.

**Comment by yoshuawuyts** [comment]
@soyuka thanks for sharing; it's very helpful!

>  (...) as rustls isn't really stable it may be more work in the future to keep this library up to date.

It seems this is the main tradeoff. RusTLS is newer, so it might require more work to keep up to date. But it comes with the upside that it's easier to run cross-platform.

I think that's an alright tradeoff actually. Like you point out the scope of our use here is reasonably small, which should limit the amount of work needed per upgrade. But it also gives us an opportunity to invest in a future we *want* to see, rather than a project that's probably not going to improve much going forward.
